### PR TITLE
expose diskus internals as library

### PR DESCRIPTION
**Pull request by amilajack**
Closes https://github.com/sharkdp/diskus/issues/12

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+extern crate humansize;
+extern crate num_cpus;
+extern crate rayon;
+
+pub mod walk;
+
+pub use walk::Walk;
+use humansize::{file_size_opts, FileSize};
+
+pub fn print_result(size: u64) {
```
Comment (sharkdp) [inline src/lib.rs]: I don't think the `print_result` function should be part of the library. This is very much application specific.

File: src/walk.rs [hunk 2]
```diff
@@ -61,6 +61,20 @@ fn walk(tx: channel::Sender<Message>, entries: &[PathBuf]) {
...
 }
 
+/// Basic usage of diskus library
+///
+/// # Example:
+///
+/// ```rust
+/// fn foo() {
+///     let mut path = PathBuf::new();
+///     path.push(".");
```
Comment (sharkdp) [inline src/walk.rs]: This could be:
```rust
let path = PathBuf::from("/");
```

File: src/walk.rs [hunk 3]
```diff
@@ -61,6 +61,20 @@ fn walk(tx: channel::Sender<Message>, entries: &[PathBuf]) {
...
+///     let paths = vec![];
```
Comment (sharkdp) [inline src/walk.rs]: Don't you have to fill that with a path?

```
let paths = vec![PathBuf::from(".")];
```
?

File: src/walk.rs [hunk 4]
```diff
@@ -61,6 +61,20 @@ fn walk(tx: channel::Sender<Message>, entries: &[PathBuf]) {
...
+///     let walk = Walk::new(&paths, 4);
+///     let size = walk.run();
+///     print_result(size);
```
Comment (sharkdp) [inline src/walk.rs]: I'd remove the `print_result` part here.

**Comment by sharkdp** [comment]
Thank you very much for working on this!

**Comment by amilajack** [comment]
Addressed comments

**Comment by sharkdp** [comment]
Thank you for the quick update!

Please don't rebase & force-push when addressing review comments. I know you probably just want keep the commit history clean (which is nice), but it's hard to review the updates because I cannot see the incremental changes.

The doc-comment has not been moved, right? Also, can we please remove the `fn foo() { ..` part?

**Comment by amilajack** [comment]
What are you referring to when you say `doc-comment`? Are you referring to the `fn foo() {}`?

**Comment by sharkdp** [comment]
Yes, the "basic usage" documentary comment that you wrote.

**Comment by amilajack** [comment]
Got it. Removed it

**Comment by sharkdp** [comment]
I think we should move it to `lib.rs` (as posted above), not remove it. I just wanted to remove the dummy function, but keep the function body. sorry for the confusion.

**Comment by amilajack** [comment]
Oh got it. My bad, read 'move' as 'remove'. Fixed.

**Comment by sharkdp** [comment]
Thank you

**Pull request merged**